### DISCUSSION TITLE
I download 2.44 to install on my mac. It crashes when opening.

### DISCUSSION THREAD

**Discussion by troach242** [christofmuc/KnobKraft-orm#364] [General]
Something may have gone wrong in when creating the dng file for this release. I downloaded it, moved it to my applications folder and launched. The message says KnobKraft Arm 2.4.4 quit unexpectedly. Thought I would let you know. :) 

<img width="535" alt="Screenshot 2024-11-06 at 4 01 19 PM" src="https://github.com/user-attachments/assets/658eefa6-e436-4a57-a1b9-8120e25df5e2">

**Comment by christofmuc** [comment]
Ouch. Can you check the console app, it should have a more detailed crash report?

I will also test on my M1.

**Reply by troach242** [reply]
2.4.3 has the same issue. I just opened 2.4.4 again and here is the crash report.
[KnobKraftOrm 2.4.4 Crash Report MACOS Intel.txt](https://github.com/user-attachments/files/17654641/KnobKraftOrm.2.4.4.Crash.Report.MACOS.Intel.txt)

**Reply by troach242** [reply]
2.4.0 is working correctly on my mac. :) I hope is not too big of an issue. Thank you for the quick response and attention.

**Reply by christofmuc** [reply]
Confirmed, again some problems with the github Mac runner. I need to dig in what the difference between 2.4.0 and 2.4.4 build is!

Thanks for reporting!

**Reply by troach242** [reply]
Any word on when the updated mac version might be ready? I am looking forward to some of the changes you made since 2.4.0. Thanks, Thom.

**Reply by christofmuc** [reply]
Thanks for reminding me! I need to find out what is wrong with the gitlab build machines. Do you need Intel or Apple silicon?

**Reply by christofmuc** [reply]
Hm, looking at the crash dump yours looks different than mine. It crashes somewhere in the initialization of an adaptation, sadly we don't know which. 

Can you try:
* Make sure you have Python 3.12 universal binary installed, you get this as an installer from python,org
* Remove all adaptations from the adaptation directory inside the Application's folder
* Start the app without Python installed by setting the environment variable ORM_NO_PYTHON to 1. 

This might be a bit harder than required because on Mac some things work differently, and I need to connect my Mac to test the same thing. Mostly I am on Windows like now, but I can help with the tests.

**Comment by troach242** [comment]
Sorry for the delay in response. I am not much of a coder. I can try to help.

* I checked my python and I have 3.12.5 installed.
* adaptations removed.. directory is blank.
* I am not sure where to find the environment variable ORM_NO_PYTHON… is that is a file somewhere?

Thanks.
Thom

**Comment by troach242** [comment]
Now that the holidays are past, I have time to work with everyone to resolve the 2.44 MACOS installation issues. I tried to get started above but need some guidence. Thanks. :)

**Reply by christofmuc** [reply]
Sorry for the delay - I spent quite some time over the holidays to get a better Mac build.

Can you try this?
https://github.com/christofmuc/KnobKraft-orm/actions/runs/12612470642/artifacts/2386113848

This is a new DMG which is changed in three ways:

1. It is a universal binary, so it has Intel as well as Apple Silicon code
2. It is signed with my developer ID, so it should not need special Open override
3. It contains the Python version as a framework, so no external Python should be needed anymore.

As this is a big change,I wouldn't be too surprised if that doesn't work either, but it launches ok on my Macbook.

**Reply by troach242** [reply]
The application opened for me and seems to be running as expected. Whatever you did to allow the app to run on Mac seems to have worked. Thank you for looking into this. 

I am looking at the features from version post 2.40 to confirm features made it into the build. 

- I checked that I can add a category and it correctly stays on the page.. yay. It was jumping back to page 1.
- Renaming a preset still creates a copy instead of renaming a preset in place.

The following may be correct behavior. I can't remember if you made changes to this part or not.
- importing a bank of presets works, but does not default to creating a BANK (not sure if that was implemented yet or if there is a config or something).
- importing a syx bank with some dups skips them.. but does not reference them as part of the import.. Ex import 100 with 20 dups only imports 80, but there is no reference to missing 20. 
The import part was something I was interested in so that when loading/importing banks they could remain a cohesive unit even if there are dups..

**Reply by christofmuc** [reply]
The two features are still open:

#351 is about including the patches in the import even if they were deduped. This requires a database change for the imports, I am halfway through it and then got distracted. 

Creating a bank during import is harder than it looks, because you might import from disk just a bunch of edit buffers or such, not necessarily forming a whole new bank. So we could create a bank on download from the synth only, which is essentially the same as #351.

**Comment by christofmuc** [comment]
Closing this as the Mac builds work again from 2.5.0 onward.
